evap/staff/tools.py

```diff
@@ -132,12 +132,14 @@ def bulk_update_users(request, user_file_content, test_run):  # noqa: PLR0912
             users_to_be_updated.append((matching_user, imported_email))
 
     emails_of_non_obsolete_users = set(imported_emails) | {user.email for user, _ in users_to_be_updated}
-    deletable_users, users_to_mark_inactive = [], []
+    deletable_users, users_to_mark_inactive, inactive_users = [], [], []
     for user in UserProfile.objects.exclude(email__in=emails_of_non_obsolete_users):
         if user.can_be_deleted_by_manager:
             deletable_users.append(user)
         elif user.is_active and user.can_be_marked_inactive_by_manager:
             users_to_mark_inactive.append(user)
+        elif not user.is_active:
```

Comment:
The elif here seems suspicious to me. An inactive user that has `can_be_marked_inactive_by_manager == False` wouldn't end up in `inactive_users`, and thus we wouldn't call `remove_inactive_participations`. I don't know whether such a user can exist, but I'd prefer to not have to think about this.

I think the loop here was meaningful and concise before, and just handled user deletion (or deactivation, if deletion isn't possible). I'd view removal of participations as orthogonal, and try to keep these as separate as possible, i.e., first just handle all deletion/marking inactive, and then handle participation archival on the remaining users. What do you think about that?